cmd/snap-update-ns/main.go

```diff
@@ -28,11 +28,13 @@ import (
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/interfaces/mount"
 	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/snap"
 )
```

Comment:
The amount of parsing of arguments in `bootstrap.c` is very limited and primitive and will allow certain things that are disallowed by `bootstrap.go`. We just check for absolutely bare essentials to prevent most obvious incorrect arguments from being passed (such as a SNAP_NAME that looks like a path).

I can adapt sc_snap_name_validate to work in `bootstrap.c` (though it would be a copy since linking this is already a bit messy) and then do as you suggested (check boostrap as early as possible). I'll also adopt the suggested comments.

As for the imports, I think we can indeed drop the 2nd validate once we do full validation in C and then try to reduce the rest as we go.